_drafts/2015-12-16-improving-scrolling-performance-of-a-UICollectionView.markdown

```diff
@@ -0,0 +1,41 @@
+
+![FPS after][2]
+[2]: http://res.cloudinary.com/wework/image/upload/v1450306072/engineering/Screen_Shot_2015-12-07_at_3.51.58_PM.png
+
+#### Additional Tips
+
+To add Re: image rendering, in addition to *Debug > Color Blended Layers*, you can use  *Debug > Color Misaligned Images* to show scaled images in yellow and misaligned images in magenta. The best bet for fast scrolling is no blending, no scaled images, no misaligned images, no shadows modulo what you can live with with respect to design. 
+
+To improve the thumbnail insertion if you have significant number of thumbnail images, consider using `NSCache/NSPurgeable` for the thumbnail images. In other words, create the thumbnail image on the fly and store the result in a cache (speculative caching) based on a key using the cell or record ID. Then the next time the image is needed, rather than recalculating/resizing you re-use the cached image. If it has been purged, you recalculate.  This is pretty straightforward and can be used for all scrolling which has a number of images which are expensive to either generate or retrieve. It is also what the Apple home screen uses for all the App Icons.
+
+There is a WWDC talk Re: [how Apple implemented the home screen scrolling](https://developer.apple.com/videos/play/wwdc2015-212/) and all of the parameters they had to consider such as cache size, memory usage vs cache usage. It is an excellent talk for understanding `NSCache/NSPurgeable`.
+
+Hope this helps.
```

Comment:
thanks Ramin for the feedback. I have incorporated your comments. Did you mean all of "Additional Tips" or the last section "There is a WWDC talk ..."?